packages/kit/src/runtime/server/page/render.js

```diff
@@ -73,6 +73,15 @@ export async function render_response({
 			: null;
 
 	if (page_config.ssr) {
+		if (!branch.at(-1)?.node.component) {
+			// Can only be the leaf, layouts have a fallback component generated
+			throw new Error(
+				`+page.svelte component for route ${event.route.id} is missing. ` +
+					`It is required because route is rendered in response to ${
+						action_result ? 'a form action' : 'a navigation'
+					}.`
+			);
+		}
```

Comment:
I'm not sure the extra context is required, we can keep the message simple I think

```suggestion
			throw new Error(`Missing +page.svelte component for route ${event.route.id}`);
```

Reply:
I was debating this myself, me more I think about the more I agree